core/src/main/scala/org/http4s/Scheme.scala

```diff
@@ -0,0 +1,59 @@
+package org.http4s
+
+import cats.{Order, Show}
+ import org.http4s.internal.parboiled2.CharPredicate.{Alpha, Digit}
+import org.http4s.internal.parboiled2.{Parser => PbParser}
+import org.http4s.parser.Http4sParser
+import org.http4s.util.{hashLower, Writer}
+
+final class Scheme private (val value: String) extends Comparable[Scheme] {
+  override def equals(o: Any) = o match {
+    case that: Scheme => this.value.equalsIgnoreCase(that.value)
+    case _ => false
+  }
+
+  private[this] var hash = 0
+  override def hashCode(): Int = {
+    if (hash == 0) {
+      hash = hashLower(value)
+    }
+    hash
+  }
+
+  override def toString = s"Scheme($value)"
+
+  override def compareTo(other: Scheme): Int =
+    value.compareToIgnoreCase(other.value)
+}
+
+object Scheme {
+  val http: Scheme = new Scheme("http")
+  val https: Scheme = new Scheme("https")
+
+  def parse(s: String): ParseResult[Scheme] =
```

Comment:
I still like `parse` because `ParseResult`, but we've argued for `fromString` here.  I could be persuaded either way.